src/actions/editorialWorkflow.js

```diff
@@ -291,3 +326,66 @@ export function publishUnpublishedEntry(collection, slug) {
     });
   };
 }
+
+export const registerUnpublishedEntryDependency = unpublishedEntryRegisterDependency;
+
+export function getUnpublishedEntryDependencies (collection, slug) {
+  return (dispatch, getState) => {
+    dispatch(unpublishedEntryDependenciesRequest(collection, slug));
+
+    const getDepsPath = dep => [
+      "entities",
+      dep,
+      "metaData",
+      "dependencies",
+    ];
+
+    const getEventualDependencies = (paths, loadedDeps, state) =>
+       paths.filter(path => !loadedDeps.includes(path)).map((path) => {
+         const [pathCollectionName, pathSlug] = path.split(".");
+         const pathCollection = state.collections.get(pathCollectionName);
+         // Wait for the entry to load
+         return dispatch(loadUnpublishedEntry(pathCollection, pathSlug))
+         // Return the path at the end so we can use it in .thens later
+           .then(() => path);
+       });
+
+    const traverse = (collectedDeps, path) => {
```

Comment:
This function closes over `getState`, which is why it was defined here. I can hoist it and make `getState` an argument, though.